src/dbnode/persist/fs/index_write.go

```diff
@@ -170,6 +175,11 @@ func (w *indexWriter) WriteSegmentFileSet(
 		minorVersion: segmentFileSet.MinorVersion(),
 		metadata:     segmentFileSet.SegmentMetadata(),
 	}
+	// We only store index segment state for snapshotted segments.
+	fstSegmentDataFileSetWriter, ok := segmentFileSet.(idxpersist.FSTSegmentDataFileSetWriter)
+	if ok {
+		seg.segmentState = fstSegmentDataFileSetWriter.SegmentState()
```

Comment:
Can we add method "SegmentState" to the interface "IndexSegmentFileSetWriter"? Would rather explicitly use the default value returned by the segment file set writer rather than use it by default if we can't upcast.